File: src/api/models/Suggestionmodel.js
const mongoose = require("mongoose");
const mongoosePaginate = require("mongoose-paginate");

const SuggestionSchema = new mongoose.Schema({
  
  countdown: { type: Number, default: 0 },
  foodId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Food",
  },
  exerciseId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Exercise",
  },
  symptoms: { type: String, default: "" },
  timeline: { type: String, default: "" },
  description: { type: String, default: "" },
});

SuggestionSchema.plugin(mongoosePaginate);

module.exports = mongoose.model("Suggestion", SuggestionSchema);

File: client/lib/src/suggestion/models/suggestion_model.dart
class Suggestion {
  final dynamic id;
  final dynamic countDown;
  final dynamic foodId;
  final dynamic exerciseId;
  final String symptoms;
  final dynamic timeline;
  final String description;

  Suggestion(
      {this.id,
      this.countDown,
      required this.foodId,
      required this.exerciseId,
      required this.symptoms,
      this.timeline,
      required this.description});

  factory Suggestion.fromJson(Map<String, dynamic> json) {
    return Suggestion(
        id: json['_id'],
        countDown: json['countdown'],
        foodId: json['foodId'],
        exerciseId: json['exerciseId'],
        symptoms: json['symptoms'],
        timeline: json['timeline'],
        description: json['description']);
  }
}
